### PR TITLE
[BEAM-5492] Python Dataflow integration tests should export the pipeline console output to Jenkins Test Result section

### PR DESCRIPTION
**Pull request by andoni-guzman**
Added some logs in dataflow runner, in case of time out or failure of jobs. looking to get more information to debug .
Added logs in dataflow runner and test dataflow runner. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by andoni-guzman** [comment]
Run Load Tests Python CoGBK Dataflow Streaming

**Comment by andoni-guzman** [comment]
Run Load Tests Python CoGBK Dataflow Streaming

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 1]
```diff
@@ -1608,6 +1608,8 @@ def is_in_terminal_state(self):
     return PipelineState.is_terminal(self._get_job_state())
 
   def wait_until_finish(self, duration=None):
+    consoleUrl = """Console URL: 'https://console.cloud.google.com/
+      dataflow/jobs/<RegionId>/{}?project=<ProjectId>'""".format(self.job_id)
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1612]: Big thank you for helping with this issue. Seeing a formatting error:

```
AssertionError: Job did not reach to a terminal state after waiting indefinitely.Console URL: 'https://console.cloud.google.com/ |  
-- | --
  | dataflow/jobs/<RegionId>/<bound method DataflowPipelineResult.job_id of <DataflowPipelineResult <Job |  
  | clientRequestId: '20220504201710949356-2017'
```
self.job_id() ?

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1612]: PTAL at other places as well. Thank you!

**Comment by tvalentyn** [comment]
Hi @andoni-guzman , is this ready for review now? Thanks!
Also, while you are looking at this, I wonder if you have an idea why certain test runs publish relevant errors in TestResults grouped by test name, while others don't? See:
https://ci-beam.apache.org/job/beam_PostCommit_Python38/2605/
vs
https://ci-beam.apache.org/job/beam_PostCommit_Python37/5191/

In the latter case, the error is buried in console log: 
```
Access Denied: Table apache-beam-testing:                python_read_table_16517355696115.python_new_types: User does not have permission to query table apache-beam-testing:                        python_read_table_16517355696115.python_new_types.
```

hunting it down adds a lot of friction.

**Comment by andoni-guzman** [comment]
Hi @tvalentyn . I think is ready, but I would like to another test using for example  Python CoGBK Dataflow Streaming to verify that now the format is correct.

Related to why in some tests publish relevant errors, in this particular case I believe it's for the kind of error, in the second one ( 5191) I saw an error that is not present in 2605 . Maybe that error invoke a log for that problem.
Also , doing some research for this task, I notice that some tests use test_dataflow_runner and other dataflow_runner. 
Test_dataflow_runner had some logs that could help. But I added that kind of logs in both files to get more info of a failure.

Yes  hunting logs adds a lot of friction, maybe we can look in a future to try to add logging.error in exceptions or something else. If you have an idea of how we can improve this, I'll with the disposition to help

**Comment by andoni-guzman** [comment]
Run Load Tests Python CoGBK Dataflow Streaming

**Comment by andoni-guzman** [comment]
Hi @tvalentyn , 
I check the last Python CoGBK Dataflow Streaming that I ran , now the format is correct. I think the PR is ready for review c:

**Comment by tvalentyn** [comment]
> Yes hunting logs adds a lot of friction, maybe we can look in a future to try to add logging.error in exceptions or something else. If you have an idea of how we can improve this, I'll with the disposition to help

I am under impression that for proper collection of logs, they need to be put into  some .xml files, like done in here: https://github.com/apache/beam/blob/178441dcf0a1e03611c46bdd12ab27907289e310/sdks/python/scripts/run_integration_test.sh#L280 . Perhaps some our tests don't do this properly?

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 2]
```diff
@@ -1628,11 +1630,13 @@ def wait_until_finish(self, duration=None):
       # is_in_terminal_state.
       terminated = self.is_in_terminal_state()
       assert duration or terminated, (
-          'Job did not reach to a terminal state after waiting indefinitely.')
+          'Job did not reach to a terminal state after waiting indefinitely.'
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1633]:
```suggestion
          'Job did not reach to a terminal state after waiting indefinitely. '
```

File: sdks/python/apache_beam/runners/dataflow/test_dataflow_runner.py [hunk 3]
```diff
@@ -84,7 +86,10 @@ def build_console_url(self, options):
 
   def wait_until_in_state(self, expected_state, timeout=WAIT_IN_STATE_TIMEOUT):
     """Wait until Dataflow pipeline enters a certain state."""
+    consoleUrl = """Console URL:https://console.cloud.google.com/dataflow/jobs/
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/test_dataflow_runner.py:89]:
```suggestion
    consoleUrl = """Console URL: https://console.cloud.google.com/dataflow/jobs/
```

**Review by tvalentyn — APPROVED** [review]

**Comment by tvalentyn** [comment]
> It seems that the python precommit is not healthy - is that independent of this PR?

I am not seeing these errors on other PRs, @andoni-guzman could you please verify if it's related and if not file a JIRA? Thanks!

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 4]
```diff
@@ -1611,7 +1611,9 @@ def wait_until_finish(self, duration=None):
     if not self.is_in_terminal_state():
       if not self.has_job:
         raise IOError('Failed to get the Dataflow job id.')
-
+      consoleUrl = """Console URL: 'https://console.cloud.google.com/
+          dataflow/jobs/<RegionId>/{}?project=
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py:1615]: These empty spaces will be concatenated into triple-quoted string. Use + or ('literal 1 with space at the end ' 'literal 2 ' ...)

File: sdks/python/apache_beam/runners/dataflow/test_dataflow_runner.py [hunk 5]
```diff
@@ -84,7 +86,10 @@ def build_console_url(self, options):
 
   def wait_until_in_state(self, expected_state, timeout=WAIT_IN_STATE_TIMEOUT):
     """Wait until Dataflow pipeline enters a certain state."""
+    consoleUrl = """Console URL: https://console.cloud.google.com/dataflow/jobs/
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/test_dataflow_runner.py:89]: once PR is ready for review, please post PTAL or a similar comment.

**Comment by andoni-guzman** [comment]
Run Load Tests Python CoGBK Dataflow Streaming

**Comment by andoni-guzman** [comment]
Run Load Tests Python CoGBK Dataflow Streaming

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**